tests/memory_checker/test_memory_checker.py

```diff
@@ -168,152 +202,116 @@ def restart_monit_service(duthost):
 
-    Args:
-        duthost: The AnsibleHost object of DuT.
-        container_name: A string represents the name of container.
+    yield
 
-    Returns:
-        None.
-    """
-    logger.info("Removing 'stress' utility from '{}' container ...".format(container_name))
-    remove_cmd_result = duthost.shell("docker exec {} apt-get purge stress -y".format(container_name))
-    exit_code = remove_cmd_result["rc"]
-    pytest_assert(exit_code == 0, "Failed to remove 'stress' utility!")
-    logger.info("'stress' utility was removed.")
+    restore_monit_config_files(duthost)
+    restart_monit_service(duthost)
 
-    logger.info("Bringing up all BGP sessions ...")
-    duthost.shell("config bgp startup all")
-    logger.info("BGP sessions are started up.")
+    if not container.is_running():
+        container.restart()
+    container.post_check()
 
 
 @pytest.fixture
-def test_setup_and_cleanup(duthosts, creds, enum_dut_feature_container,
-                           enum_rand_one_per_hwsku_frontend_hostname, request):
-    """Backups Monit configuration files, customizes Monit configuration files and
-    restarts Monit service before testing. Restores original Monit configuration files
-    and restart Monit service after testing.
+def remove_and_restart_container(memory_checker_dut_and_container):
+    """Removes and restarts 'telemetry' container from DuT.
 
     Args:
-        duthost: Hostname of DuT.
+        memory_checker_dut_and_container: Fixture providing the duthost and container to test
 
     Returns:
         None.
     """
-    dut_name, container_name = decode_dut_and_container_name(enum_dut_feature_container)
-    pytest_require(dut_name == enum_rand_one_per_hwsku_frontend_hostname,
-                   "Skips testing memory_checker of container '{}' on the DuT '{}' since another DuT '{}' was chosen."
-                   .format(container_name, dut_name, enum_rand_one_per_hwsku_frontend_hostname))
-
-    pytest_require(container_name == "telemetry",
-                   "Skips testing memory_checker of container '{}' "
-                   "since memory monitoring is only enabled for 'telemetry'."
-                   .format(container_name))
-
-    duthost = duthosts[dut_name]
-
-    install_stress_utility(duthost, creds, container_name)
-
-    backup_monit_config_files(duthost)
-    customize_monit_config_files(duthost, request.param)
-    restart_monit_service(duthost)
+    duthost, container = memory_checker_dut_and_container
+    container.remove()
 
     yield
 
-    try:
-        restore_monit_config_files(duthost)
-    finally:
-        restart_monit_service(duthost)
+    if not container.is_running():
+        container.restart()
+    container.post_check()
 
-        restart_container(duthost, container_name)
-        remove_stress_utility(duthost, container_name)
-        postcheck_critical_processes(duthost, container_name)
+
+@pytest.fixture(params=['telemetry'])
+def enum_memory_checker_container(request):
```

Comment:
sorry I missed review this part. could we please abandon this fixture?
we have existing fixture in conftest.py to get feature on the enum dut. I have merged https://github.com/sonic-net/sonic-mgmt/pull/9218.

Reply:
@wenyiz2021 I could do this but however note:

1) I would have to bring ` pytest_require(container_name == "telemetry"` back in the code. 
This ends up skipping the test on all containers and just enables it on telemetry.
This ends up increasing the runtime of the test for a similar end goal.
And polluting the test results with tons of skip.

2) It personally feels useless to test this feature on more than just 1 container.
This test is pretty much checking that monit works as expected, It bears no additional benefit to run this on all container. Bear in mind that these 4 tests take 15.
So if we do this times the number of containers we're looking at 2 hours which seems really wasteful.

3) As of today only 2 containers support this memory check `telemetry` and `snmp`.
Both of these are in the default namespace in multi asic devices.
Also if we were to enable this memory check on `swss` or other critical containers, the test would need to be redesigned since it would not survive more than 1 container flapping and containers taking more than 60 seconds to come back up.

In my opinion we should keep this fixture, it's cleaner that the `pyrequire` that's currently there and wasting time since it has to iterate over all the containers. 
I don't want to waste cycles changing the code to revert it back later, so I want to hear back from you if that's really the path we want to go down.
cc @aaronpayment

Reply:
@Staphylo I agree with you, we could use `@pytest.fixture(params=['telemetry'])` iso skip all containers that are not telemetry